doc/faq.markdown

```diff
@@ -1,11 +1,19 @@
 Frequently Asked Questions {#faq}
 ==========================
 
--   Q: Example question?
+-   **What is InputArray?**
 
-    A: Example answer
+    It can be seen that almost all OpenCV functions receive InputArray type. What is it, and how can I understand the actual input types of parameters?
 
 
--   Q: Example question?
+    This is the proxy class for passing read-only input arrays into OpenCV functions.
 
-    A: Example answer
+    Inside a function you should use _InputArray::getMat() method to construct a matrix header for the array (without copying data). _InputArray::kind() can be used to distinguish Mat from vector<> etc., but normally it is not needed.
+
+    for more information see cv::InputArray
+
+-   **Which is more efficient, use contourArea() or count number of ROI non-zero pixels?**
+
+    In a case were you only want relative areas, which one is faster to compute: calculate a contour area - contourArea() - or count the number of ROI non-zero pixels?
```

Comment:
@csukuangfj thank you. i just copied questions and answers from answers.opencv.org